### PR TITLE
File: Used emit instead of push in onUpstreamFinish

### PR DESCRIPTION
**Pull request by michalbogacz**
##  Purpose

Fix for possible error "Cannot push port ZipArchiveFlow.out twice"

## References

References #2010

## Changes

`emit` is used instead of `push` in `onUpstreamFinish()`

### CODE REVIEW

File: file/src/test/scala/akka/stream/alpakka/file/impl/archive/ZipArchiveFlowTest.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.Keep
+import akka.stream.testkit.scaladsl.{TestSink, TestSource}
+import akka.testkit.TestKit
+import akka.util.ByteString
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{Matchers, WordSpecLike}
+
+class ZipArchiveFlowTest extends TestKit(ActorSystem("ziparchive")) with WordSpecLike with Matchers with ScalaFutures {
```
Comment (ennru) [inline file/src/test/scala/akka/stream/alpakka/file/impl/archive/ZipArchiveFlowTest.scala]: Please add a shutdown of the actor system.

```scala
  override def afterAll(): Unit = {
    super.afterAll()
    TestKit.shutdownActorSystem(system)
  }
```

Comment (michalbogacz) [reply file/src/test/scala/akka/stream/alpakka/file/impl/archive/ZipArchiveFlowTest.scala]: Fixed. In `ArchiveSpec` there was this issue too, so I fixed it too.

**Comment by ennru** [comment]
Thank you!

**Comment by mkurz** [comment]
Thanks a lot for fixing this @michalbogacz! #2010 bit me before christmas and now when coming back to it after a while to implement a fix I just saw you were faster. Fantastic!

Any chance you could also implement UnzipArchiveFlow :wink:?

**Comment by michalbogacz** [comment]
Thanks :slightly_smiling_face: 
Zip flow was something I needed for work. Right now I don't plan to implement Unzip flow.

**Comment by mkurz** [comment]
@michalbogacz :+1: Thanks!

**Pull request merged**